library_generation/cli/entry_point.py

```diff
@@ -88,6 +88,20 @@ def generate(
     The commit history, if generated, will be available in
     repository_path/pr_description.txt.
     """
+    generate_impl(
+        baseline_generation_config_path, current_generation_config_path, repository_path
+    )
+
+
+def generate_impl(
+    baseline_generation_config_path: str,
+    current_generation_config_path: str,
+    repository_path: str,
+):
+    """
+    implementation method for generate()
```

Comment:
```suggestion
    implementation method for generate()
    this decouples the exposed interface from the implementation
    in order to ease testing
```

Reply:
Maybe add additional info that this function is exposed for testing only and should not be called directly?

Reply:
Good point. I improved the comment.